src/bp-messages/classes/class-bp-messages-thread.php

```diff
@@ -189,20 +205,28 @@ public function populate( $thread_id = 0, $order = 'ASC', $args = array() ) {
 			return false;
 		}
 
-		$last_message_index         = count( $this->messages ) - 1;
-		$this->last_message_id      = $this->messages[ $last_message_index ]->id;
-		$this->last_message_date    = $this->messages[ $last_message_index ]->date_sent;
-		$this->last_sender_id       = $this->messages[ $last_message_index ]->sender_id;
-		$this->last_message_subject = $this->messages[ $last_message_index ]->subject;
-		$this->last_message_content = $this->messages[ $last_message_index ]->message;
+		// Messages count.
+		$this->messages_count = self::get_total_thread_message_count( $this->thread_id );
 
-		foreach ( (array) $this->messages as $key => $message ) {
-			$this->sender_ids[ $message->sender_id ] = $message->sender_id;
+		// Sender IDs.
+		$this->sender_ids = self::get_thread_sender_ids( $this->thread_id );
+
+		// Get and set last message.
+		$message = self::get_latest_thread_message( $this->thread_id );
+		if ( ! empty( $message ) && is_object( $message ) ) {
+			$this->last_message_id      = $message->id;
+			$this->last_message_date    = $message->date_sent;
+			$this->last_sender_id       = $message->sender_id;
+			$this->last_message_subject = $message->subject;
+			$this->last_message_content = $message->message;
 		}
 
 		// Fetch the recipients and set the displayed/logged in user's unread count.
 		$this->recipients = $this->get_recipients( $thread_id, $r );
 
+		// Set total recipients count.
+		$this->recipients_count = self::get_total_thread_recipients_count( $this->thread_id );
```

Comment:
Another query, I guess it's still a downsize of pagination?

Reply:
No, this is a new piece of data that I'm introducing. The Slack app is a good example of how this information might be useful. Currently, we return the recipients `array`, but since it is paginated, the user never gets the total number of recipients. They need to request all and calculate from that.